packages/open-next/src/adapters/routing/middleware.ts

```diff
@@ -137,8 +137,15 @@ export async function handleMiddleware(
   // NOTE: the header was added to `req` from above
   const rewriteUrl = responseHeaders.get("x-middleware-rewrite");
   let rewritten = false;
+  let middlewareQueryString = internalEvent.query;
   if (rewriteUrl) {
-    req.url = new URL(rewriteUrl).pathname;
+    const rewriteUrlObject = new URL(rewriteUrl);
+    req.url = rewriteUrlObject.pathname;
+    rewriteUrlObject.searchParams.forEach((v: string, k: string) => {
+      if (!middlewareQueryString[k]) {
```

Comment:
With a defensive approach, I have considered that if, for some reason, the internalUrl already contains a field with that key set, it should take precedence over its counterpart set during the rewrite phase. However, if that's not the case, I can remove the if statement that checks this condition.

Reply:
I'm not sure if there's a reserved parameter that the internal event requires, if there is, we should try to find it... otherwise it makes more sense to use the rewriteUrl's params.

Reply:
I agree, I've removed the if statement to allow full usage of the rewriteUrl's params

Reply:
This doesn't seem right to me, we should pass the initial search param if no search params is defined in middleware (And even that seems off) and just pass the ones defined in middleware if people have defined some.
People have access to the request in middleware, so if they want to forward the params it's up to them.
What if someone want to remove those params, it will be impossible.

Reply:
Personally i'd say, it's up to the user, if they do a rewrite they can add their own search params and even reuse ones from initial request. It should be entirely up to them.
Is there a case where we'd want to force some search params ? @khuezy @pbaire

Reply:
Hi @conico974,

I apologize for the delay in my response; I've been quite busy these past few days. 
I agree with you that people have access to the request in middleware, but I'm having trouble figuring out how to add my own search parameters without making this changes to the code. You can see my initial attempt in the first commit of the pull request, where I set u.searchParams.set("a", "b");.

So, my question is, how can I use the search parameters when performing a rewrite to a specific URL, or is this considered bad practice?

Reply:
No there is some change necessary to make this work. 
What i think is not good is that you reuse the `internalEvent.query`, if there is a rewrite we should use only the search params set in middleware and not the initial ones. I'd do something like that instead 
```ts
if (rewriteUrl) {
    const rewriteUrlObject = new URL(rewriteUrl);
    req.url = rewriteUrlObject.pathname;
    middlewareQueryString = rewriteUrlObject.searchParams;
    rewritten = true;
  }
```

Reply:
@conico974 now I got it, I've pushed another commit for use only the search params set in middleware and not the initial ones

Reply:
The original code uses: `query: internalEvent.query,`, was that a bug before?

Reply:
@khuezy  sorry I forgot that in the original code we pass the internalEvent.query  by default,
so I've restored this default and, if a middleware rewrite is present we reset with a brand new set of parameters for use only the search params set in middleware and not the initial ones as @conico974 requested